examples/js_callback/src/main.rs

```diff
@@ -1,77 +1,88 @@
 use wasm_bindgen::prelude::*;
-use web_sys::HtmlTextAreaElement;
+use wasm_bindgen::JsCast;
+use wasm_bindgen_futures::JsFuture;
+
+use once_cell::sync::OnceCell;
 use yew::prelude::*;
+use yew::suspense::{use_future, SuspensionResult};
 
 mod bindings;
 
-pub enum Msg {
-    Payload(String),
-    AsyncPayload,
+static WASM_BINDGEN_SNIPPETS_PATH: OnceCell<String> = OnceCell::new();
+
+#[function_component]
+fn Important() -> Html {
+    let msg = use_memo(|_| bindings::hello(), ());
+    html! {
+        <>
+            <h2>{"Important"}</h2>
+            <p>{msg}</p>
+        </>
+    }
+}
+
+#[wasm_bindgen]
+extern "C" {
+    type Module;
+
+    #[wasm_bindgen(method)]
+    fn bye(this: &Module) -> String;
```

Comment:
I think it'd be cleaner if you moved all the `wasm_bindgen` stuff into the `bindings` module so that it exports only (roughly, not actual syntax). But it entails moving `WASM_BINDGEN_SNIPPETS_PATH` and encapsulating the loading in `use_do_bye`.

```
mod bindings {
    fn hello() -> String;
    pub type UnimportantModule; // Module -> UnimportantModule
    impl UnimportantModule {
        // via #[wasm_bindgen(method)]
        pub fn bye(&self) -> String;
    }
    async fn import_unimportant() -> UnimportantModule;
}
```

Reply:
I decided to keep `WASM_BINDGEN_SNIPPETS_PATH` in main.rs but moved the `UnimpModule` to bindings module